### ISSUE TITLE
Feature request: arbitrary structure for input, and output original

### ISSUE THREAD

**Issue by Kitanotori** [manuelbieh/geolib#127]
Problem 1.
=======
`findNearest`, `orderByDistance`, etc. force to use specific parameter names for coordinates (`latitude`, `longitude`), which means that I have to transform all input that I fetch e.g. from [Google](https://github.com/googlemaps/google-maps-services-js), which outputs nested `lat` and `lng`.

Solution 1.
=======
Allow user to specify _how_ the coordinate parameters are named and _where_ they are stored in the object, i.e. allow using arbitrary named nested parameters, for example by utilizing lodash's `_.get()`.

Problem 2.
=======
`findNearest`, `orderByDistance`, etc. do not return the original input.

Solution 2.
=======
Return the original input for example in following structure:

```
[{
  ...nearest_original_object,
  location: {
    latitude: ...,
    longitude: ...,
    distance: ...
  }
},{
  ...second_nearest_original_object,
  location: {
    latitude: ...,
    longitude: ...,
    distance: ...
  }
},...
]
```

**Comment by manuelbieh** [comment]
Hi,

that's on my list for Geolib v3 and it's already half-implemented.

But using with Google Maps objects shouldnt be a problem as I allowed some aliases for coordinate keys which also includes `lat` and `lng` (see https://github.com/manuelbieh/Geolib/blob/master/src/geolib.js#L117).

**Comment by manuelbieh** [comment]
This will be solved in v3.0.0 which will finally be released shortly! The code itself is done already, I only need to update docs, changelog, release workflows, etc ...
